vendor/src/github.com/docker/libnetwork/endpoint.go

```diff
@@ -303,6 +304,24 @@ func (ep *endpoint) isAnonymous() bool {
 	return ep.anonymous
 }
 
+// CompareAndSwap ep's serviceEnabled. If its in oldState, set it to newState
+// and return true.  If its not in oldState return false
+func (ep *endpoint) casServiceEnabled(oldState, newState bool) bool {
```

Comment:
@sanimej It's weird code and doesn't really achieve what you think it achieves. You get the same exact result if you have a single argument, since both use the complement as the `oldValue`. I have no clue why you would ever make them idempotent. Enable/Disable being a one way operation is an implementation detail of swarmkit. Users of the API should be able to add and remove the service at their leisure. Enforcing it at this level is error prone, buggy and creates weird code like this.